backend/src/graphql/resolvers/images/imagesS3.spec.ts

```diff
@@ -0,0 +1,409 @@
+          })
+          // eslint-disable-next-line no-catch-all/no-catch-all
+        } catch (err) {
+          // nothing has been deleted
+          await expect(neode.all('Image')).resolves.toHaveLength(1)
+          // all good
+        } finally {
+          await session.close()
+        }
+      })
+    })
+  })
+})
+
+describe('mergeImage', () => {
+  const { mergeImage } = images(config)
+  let imageInput: ImageInput
+  let post: { id: string }
+  beforeEach(() => {
+    imageInput = {
+      alt: 'A description of the new image',
+    }
+  })
+
+  describe('given image.upload', () => {
+    beforeEach(() => {
+      const createReadStream: FileUpload['createReadStream'] = (() => ({
+        pipe: () => ({
+          on: (_, callback) => callback(),
+        }),
+      })) as unknown as FileUpload['createReadStream']
+      imageInput = {
+        ...imageInput,
+        upload: Promise.resolve({
+          filename: 'image.jpg',
+          mimetype: 'image/jpeg',
+          encoding: '7bit',
+          createReadStream,
+        }),
+      }
+    })
+
+    describe('on existing resource', () => {
+      beforeEach(async () => {
+        const p = await Factory.build(
+          'post',
+          { id: 'p99' },
+          {
+            author: Factory.build('user', {}, { avatar: null }),
+            image: null,
+          },
+        )
+        post = await p.toJson()
+      })
+
+      it('returns new image', async () => {
+        await expect(
+          mergeImage(post, 'HERO_IMAGE', imageInput, { uploadCallback, deleteCallback }),
+        ).resolves.toMatchObject({
+          url: expect.any(String),
+          alt: 'A description of the new image',
+        })
+      })
+
+      it('calls upload callback', async () => {
+        await mergeImage(post, 'HERO_IMAGE', imageInput, { uploadCallback, deleteCallback })
+        expect(uploadCallback).toHaveBeenCalled()
+      })
+
+      it('creates `:Image` node', async () => {
+        await expect(neode.all('Image')).resolves.toHaveLength(0)
+        await mergeImage(post, 'HERO_IMAGE', imageInput, { uploadCallback, deleteCallback })
+        await expect(neode.all('Image')).resolves.toHaveLength(1)
+      })
+
+      it('creates a url safe name', async () => {
+        if (!imageInput.upload) {
+          throw new Error('Test imageInput was not setup correctly.')
+        }
+        const upload = await imageInput.upload
+        upload.filename = '/path/to/awkward?/ file-location/?foo- bar-avatar.jpg'
+        imageInput.upload = Promise.resolve(upload)
+        await expect(
+          mergeImage(post, 'HERO_IMAGE', imageInput, { uploadCallback, deleteCallback }),
+        ).resolves.toMatchObject({
+          url: expect.stringMatching(
+            new RegExp(`^http://your-objectstorage.com/bucket/${uuid}-foo-bar-avatar.jpg`),
+          ),
+        })
+      })
+
+      describe('but given a `S3_PUBLIC_GATEWAY` configuration', () => {
```

Comment:
```suggestion
      describe('given a `S3_PUBLIC_GATEWAY` configuration', () => {
```

Reply:
```suggestion
      describe('with a `S3_PUBLIC_GATEWAY` configuration', () => {
```

Reply:
Sure, there is no test case before that explicitly says what happens when you don't have it set.